nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/state/flow/flow.effects.ts

```diff
@@ -3153,4 +3157,41 @@ export class FlowEffects {
             ),
         { dispatch: false }
     );
+
+    moveToFront$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(FlowActions.moveToFront),
+            map((action) => action.request),
+            concatLatestFrom((request) => this.store.select(selectMaxZIndex(request.componentType))),
+            switchMap(([request, maxZIndex]) => {
+                const updateRequest: UpdateComponentRequest = {
+                    id: request.id,
+                    type: request.componentType,
+                    uri: request.uri,
+                    payload: {
+                        revision: request.revision,
+                        disconnectedNodeAcknowledged: this.clusterConnectionService.isDisconnectionAcknowledged(),
+                        component: {
+                            id: request.id,
+                            zIndex: maxZIndex + 1
```

Comment:
Should we compare the max with the zIndex of the selection? If they are the same, don't bother to increment?